### PR TITLE
Round of Float and small change in str.py 

### PR DESCRIPTION
**Pull request by Shikhar1998**
1. Rounding of Float
2. Correcting an except

### CODE REVIEW

File: sympy/core/numbers.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from __future__ import print_function, division
 
 import decimal
+from decimal import *
```
Comment (asmeurer) [inline sympy/core/numbers.py]: Do not use `import *`. `decimal` is already imported in the line above this. You could just use that.

File: sympy/core/numbers.py [hunk 2]
```diff
@@ -417,8 +418,11 @@ def __rdivmod__(self, other):
             raise TypeError(msg % (type(other).__name__, type(self).__name__))
         return divmod(other, self)
 
-    def __round__(self, *args):
-        return round(float(self), *args)
+    def __round__(self,ndigits = 0):
+        self = str(self)
+        index = self.index(".")
+        self = self[:index+1+ndigits]
+        return Decimal(self)
```
Comment (asmeurer) [inline sympy/core/numbers.py]: `__round__` should return a `Float`, not a `Decimal`.

Comment (asmeurer) [inline sympy/core/numbers.py]: Please use the `Float` algorithms, not the decimal ones. `Float.evalf` is (almost) what you need here.

File: sympy/core/numbers.py (hunk 1 above)
Comment (Shikhar1998) [inline sympy/core/numbers.py]: ok 
👍

**Comment by Shikhar1998** [comment]
@asmeurer I have used Float(self) instead

**Comment by Shikhar1998** [comment]
@asmeurer i have tried running it and i am getting the answer for `print __round__(pi.evalf(100),49)

print (str(pi.evalf(50)))` as
`3.1415926535897932384626433832795028841971693993751
3.1415926535897932384626433832795028841971693993751` respectively . 

*I also have a question like in Float.evalf() *
`N(sqrt(2)*pi, 5) 
==> 4.4429`
Returns a value upto 4 decimal precesion on input of 5 , is it a bug or by design ??

**Comment by Shikhar1998** [comment]
I have made the changes in test_utils.py please check

**Comment by asmeurer** [comment]
The argument of evalf/N is the total number of digits, whereas the argument of round is the number of digits after the decimal. Hence 3.14 is 3 digits but has only 2 digits after the decimal.

**Comment by Shikhar1998** [comment]
Please check test_utils.py in the optics module last test case has an assertion answer == 9.47 but the actual answer is 9.46969696969697 which is rounded as 9.47 by the round function. Shouldn't it be 9.46 please check?
I know with the the present rounding method it should be 9.47 but according to the issue #12056 updates were required.
This is the reason why my pull request is failing , Please check....:D

**Pull request closed without merging**